### PR TITLE
New get started page to have users follow libraries after signup

### PR DESCRIPTION
**Pull request by jaredpetker**
@cvializ @andrewconner Still waiting on tracking from jen, but feel free to give it a look over, mostly good to go

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.styl [hunk 1]
```diff
@@ -0,0 +1,181 @@
+kf-center-vertical(b)
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.styl]: I noticed this isn't used anywhere, it's probably best to remove it

File: kifi-backend/modules/shoebox/angular/src/home/home.styl [hunk 2]
```diff
@@ -26,7 +27,9 @@
     max-width: none
 
 .kf-home-sidebar
+  background-color: red
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/home/home.styl]: was this intentional?

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      var ids = selected.map(function(lib) {
+        return lib.id;
+      });
+
+      $scope.viewState = VIEW_STATES.FOLLOWING;
+      libraryService.joinLibraries(ids)
+          .then(function() {
+            profileService.savePrefs({has_seen_ftue: true});
+            $state.go('home.feed');
+          }, function() {
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js]: for promises it's best to use `.catch()` over `then(function(){},function(){})`. The linter complains about using the reserved word `catch` so we have to do

```
service.returnsPromise()
.then(function () {/* ... */})
['catch'](function () {/* ... */});
```

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.styl (hunk 1 above)
Comment (jaredpetker) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.styl]: Yea it's not needed now, moved the screen to flex

File: kifi-backend/modules/shoebox/angular/src/home/home.styl (hunk 2 above)
Comment (jaredpetker) [inline kifi-backend/modules/shoebox/angular/src/home/home.styl]: good catch, nope, left from debugging some stuff

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        FOLLOWING: 'FOLLOWING'
+    };
+    $scope.VIEW_STATES = VIEW_STATES;
+
+    $scope.viewState = VIEW_STATES.LOADING;
+    $scope.libraries = [];
+    $scope.selectedLibraries = [];
+    $scope.fetchLibraries = function () {
+
+      function librarySource() {
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js]: This is more of a nitpick, but for efficiency `librarySource` should be declared outside of `fetchLibraries`. How it is now the interpreter has to re-construct `librarySource` during every call to `fetchLibraries`.

**Comment by cvializ** [comment]
awesome!

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js (hunk 4 above)
Comment (jaredpetker) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js]: I see, makes sense, in general this thing would be called once for this page anyways so seems like it wouldn't make a difference, can move it for consistency though

Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js]: Oh gotcha. If it's only getting called once, you don't need the
`Paginator`/`lazyLoader` then. You can just call
`libraryService.getFtueLibraries()` directly in `fetchLibraries`. I guess
lazyLoader is sort of a misnomer on my part.

File: kifi-backend/modules/shoebox/angular/src/common/directives/simpleLoadingSpinner/simpleLoadingSpinner.styl [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  vertical-align: middle
+
+  svg
+    width: auto
+    height: auto
+    fill: accentGreen
+    animation: pulse 1.1s infinite linear
+
+@keyframes pulse
+{
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/common/directives/simpleLoadingSpinner/simpleLoadingSpinner.styl]: You can get rid of these `{` guys. Not needed in Stylus unless you love them

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.styl [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+
+.kf-ftue-fl-checkbox-wrap
+  margin-right: 15px
+
+.kf-ftue-fl-checkbox
+  display: none
+
+.kf-ftue-fl-checkbox + label
+  //background: url(/img/symbol-sprites/lib/checkbox_unchecked.svg) no-repeat
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.styl]: Needed?

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.styl [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+  width: 24px
+  display:inline-block
+  align-self: center
+
+
+.kf-ftue-fl-checkbox-bg
+  background-size: contain
+  background-repeat: no-repeat
+
+
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.styl:164]: \n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.tpl.html [hunk 8]
```diff
@@ -0,0 +1,68 @@
+<div class="kf-ftue-fl-container">
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.tpl.html:1]: You may want to change your editor, or convince/talk to Carlos. Currently we use 2 spaces for indention. :bomb:

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js (hunk 3 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js]: We can turn this off in the linter if you want

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js (hunk 4 above)
Comment (jaredpetker) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js]: for shore

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.tpl.html (hunk 8 above)
Comment (jaredpetker) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.tpl.html:1]: fixed intellij

Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibraries.tpl.html:1]: :clap: two-space-indent ftw

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js (hunk 3 above)
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js]: eh I'm used to it, I like it now even though at first it's counterintuitive

**Comment by andrewconner** [comment]
:+1: :+1: :+1: :+1: :+1: :+1:

File: kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js [hunk 9]
```diff
@@ -226,12 +225,23 @@ angular.module('kifi')
     }
 
     var unregisterAutoShowGuide = $rootScope.$watch(function () {
-      return Boolean(installService.installedVersion && profileService.prefs.auto_show_guide);
+      var keys = Object.keys(profileService.prefs);
+      return Boolean(installService.installedVersion
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: Shouldn't we show welcome even if the user didn't install?

Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: A sketch:

```
$rootScope.$watch(function () {
  var newInstall = installService.installedVersion && profileService.auto_show_guide;
  var ftue = profileService.prefs.has_seen_ftue === false;
  return newInstall || ftue;
}, function () {
  if (profileService.prefs.has_seen_ftue === false) { ... } else if (profileService.prefs.auto_show_guide) { ... }
});
```

@cvializ what do you think?

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+      trackPageClick({
+        action: 'follow',
+        num_libraries_followed: ids.length + ''
+      });
+
+      $scope.viewState = VIEW_STATES.FOLLOWING;
+      libraryService.joinLibraries(ids)
+          .then(function() {
+            profileService.savePrefs({has_seen_ftue: true});
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js:69]: @JRuff42 Do you want a "skip" link if a user doesn't want to follow libraries?

**Comment by andrewconner** [comment]
Nice job Jared

File: kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js (hunk 9 above)
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: lgtm

Comment (jaredpetker) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js]: fixed it up

File: kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+        if (!checked) {
+          $scope.selectAll = false;
+        }
+      };
+
+      $scope.selectAll = false;
+
+
+      // preload checkboxes
+      var preload = function () {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js]: Maybe I missed this... throw it in the spritesheet?

Comment (jaredpetker) [inline kifi-backend/modules/shoebox/angular/src/ftue/ftueFollowLibrariesCtrl.js]: Yea it's already there, I can remove this now

**Comment by jaredpetker** [comment]
@cvializ @andrewconner if all seems ok I'll probably wait until monday and merge it in the morning

File: kifi-backend/modules/shoebox/angular/src/home/home.styl [hunk 12]
```diff
@@ -13,6 +13,7 @@
 
 .kf-home-items
   max-width: 600px
+  height: 100px
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/home/home.styl:16]: @jaredpetker I just noticed there's a bug here. Setting this height is breaking the home screen on narrow browser widths. The sidebar is rendered behind the feed items. Does your code still work with the `.kf-home-items` and `.kf-home-sidebar` height values removed?

**Pull request merged**